src/peft/tuners/lora.py

```diff
@@ -1167,3 +1172,58 @@ def forward(self, x: torch.Tensor):
                         )
                     result += output
                 return result
+
+
+# Ugly solution
+if is_auto_gptq_available():
+
+    class QuantLinear(torch.nn.Module, LoraLayer):
+        def __init__(
+            self,
+            adapter_name,
+            quant_linear_module,
+            r: int = 0,
+            lora_alpha: int = 1,
+            lora_dropout: float = 0.0,
+            **kwargs,
+        ):
+            torch.nn.Module.__init__(self)
+            LoraLayer.__init__(
+                self, in_features=quant_linear_module.infeatures, out_features=quant_linear_module.outfeatures
+            )
+            self.weight = quant_linear_module.qweight
```

Comment:
this `self.weight` is never used. Good to remove it

Reply:
removed =)